data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java

```diff
@@ -89,4 +98,12 @@ private File getVectorLayerFile(VectorLayer vectorLayer) {
         }
         return vectorLayers;
     }
+
+    private File[] getSimilarCacheFiles(VectorLayer vectorLayer) {
+        final Pattern p = Pattern.compile(mLayerFilenameSerializer.getFilePattern(vectorLayer));
+        File f = getVectorLayerFile(vectorLayer);
+        return f
+                .getParentFile()
```

Comment:
you don't need f and getParentFile, it's a weird way to get the parent dir. you already have it:
mExternalVectorLayersDir